Sources/SwiftPhoenixClient/Socket.swift

```diff
@@ -157,6 +157,9 @@ public class Socket: PhoenixTransportDelegate {
   
   /// True if the Socket closed cleaned. False if not (connection timeout, heartbeat, etc)
   var closeWasClean: Bool = false
+
+  /// True if the Socket is being closed abnormally
+  var closingAbnormally: Bool = false
```

Comment:
Sort of seems like these should be combined in either an enum or optional bool, because it makes no sense to have both `closeWasClean` and `closingAbnormally` set to true.

Reply:
Thanks, yeah, I think that's a good idea -- an enum would nicely encapsulate the different close states.